src/Compiler/FSComp.txt

```diff
@@ -19,11 +19,16 @@ undefinedNamePatternDiscriminator,"The pattern discriminator '%s' is not defined
 replaceWithSuggestion,"Replace with '%s'"
 addIndexerDot,"Add . for indexer access."
 listElementHasWrongType,"All elements of a list must be implicitly convertible to the type of the first element, which here is '%s'. This element has type '%s'."
+listElementHasWrongTypeTuple,"All elements of a list must be implicitly convertible to the type of the first element, which here is a tuple of length %d of type\n    %s    \nThis element is a tuple of length %d of type\n    %s    \n"
```

Comment:
To me it would read better (even though it's longer) if it was
> ... which in this case is a ...

rather than

> ... which here is a ...

But I am not a native speaker, so idk 🤷
In general, are these error messages cleared with some sort of technical writer or other "expert"?

Reply:
In this case I just took the existing error messages and customized them for tuples. I don't think there is any technical writer for this, we probably just have to rely on the native speakers on the team :)

For me _here_ seems fine.